conan/tools/cmake/cmakegen.py

```diff
@@ -0,0 +1,14 @@
+from conan.tools.cmake import CMakeDeps, CMakeToolchain
+from conan.tools.env import VirtualEnv
+
+
+class CMakeGen:
```

Comment:
I am not sure, why is it needed? it seems to be only combination of 3 other generators

Reply:
It is a high level wrapper of those three. Otherwise, all recipes will have to declare ``CMakeToolchain``, ``CMakeDeps`` and ``VirtualEnv`` (or other toolchains). Its intention was to simplify the general UI for many cases.

In some other toolchains it will be more evident, like ``Autootools``. If a recipe uses ``AutooToolsToolchain``, ``AutoToolsDeps``, ``VirtualEnv``, then it will get 3 environment files, like ``buildenv.sh``, ``toolchain.sh`` and ``conandeps.sh``, and will have to activate one after the other for normal user usage. If using a higher ``AutootoolsGen`` it is possible to compose the output of the three very easily in a single ``buildenv.sh`` which will be simpler for the user.